src/screens/ChannelListScreenV2/IncognitoEmptyChat/index.tsx

```diff
@@ -31,9 +32,14 @@ const IncognitoButton = ({title, subtitle, onPress}) => {
   );
 };
 
-const IncognitoEmptyChat = ({totalChannel}) => {
-  const {goToContactScreen} = useAnonymousChannelListScreenHook();
+const IncognitoEmptyChat = () => {
+  const {channels: anonChannels, goToContactScreen} = useAnonymousChannelListScreenHook();
   const navigation = useNavigation();
+  let totalChannel = anonChannels.length;
+  const storedTotalChannel = StorageUtils.totalAnonChannels.get();
+  if (storedTotalChannel !== null && storedTotalChannel !== undefined) {
+    totalChannel = parseInt(storedTotalChannel, 10);
+  }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The changes made here are significant as they alter how `totalChannel` is calculated. Instead of receiving `totalChannel` as a prop, it's now calculated based on the length of `anonChannels` or retrieved from storage if available. This change could potentially introduce bugs if not handled correctly. It's crucial to ensure that `anonChannels` is always up-to-date and that the stored value is updated whenever `anonChannels` changes. Also, consider handling the case where `parseInt(storedTotalChannel, 10)` returns `NaN`.
```diff
- const IncognitoEmptyChat = ({totalChannel}) => {
-   const {goToContactScreen} = useAnonymousChannelListScreenHook();
+ const IncognitoEmptyChat = () => {
+   const {channels: anonChannels, goToContactScreen} = useAnonymousChannelListScreenHook();
    const navigation = useNavigation();
+   let totalChannel = anonChannels.length;
+   const storedTotalChannel = StorageUtils.totalAnonChannels.get();
+   if (storedTotalChannel !== null && storedTotalChannel !== undefined) {
+     totalChannel = parseInt(storedTotalChannel, 10);
+   }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->